### ISSUE TITLE
Panic after failed network connection

### ISSUE THREAD

**Issue by SergeyTsalkov** [go-redis/redis#172]
Apparently trying to run EvalSha after a failed network connection results in a panic. This is version 3.2.10.

```
2015/10/08 02:15:09 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:09 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:09 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:09 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:10 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:10 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:10 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:11 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:14 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:24 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:27 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:28 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:29 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:29 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
2015/10/08 02:15:29 redis: new failed: dial tcp 192.99.148.160:6379: i/o timeout
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x38 pc=0x710b44]

goroutine 204 [running]:
gopkg.in/redis%2ev3.(*singleConnPool).Get(0xc8209bcba0, 0x0, 0x0, 0x0)
	/home/sergey/go/src/gopkg.in/redis.v3/pool.go:356 +0x104
gopkg.in/redis%2ev3.(*baseClient).conn(0xc820950780, 0x7f61d2cc9900, 0x0, 0x0)
	/home/sergey/go/src/gopkg.in/redis.v3/redis.go:20 +0x45
gopkg.in/redis%2ev3.(*baseClient).process(0xc820950780, 0x7f61d2cbaee8, 0xc8207e8aa0)
	/home/sergey/go/src/gopkg.in/redis.v3/redis.go:45 +0x82
gopkg.in/redis%2ev3.(*baseClient).(gopkg.in/redis%2ev3.process)-fm(0x7f61d2cbaee8, 0xc8207e8aa0)
	/home/sergey/go/src/gopkg.in/redis.v3/redis.go:185 +0x34
gopkg.in/redis%2ev3.(*commandable).Process(0xc8207eaa38, 0x7f61d2cbaee8, 0xc8207e8aa0)
	/home/sergey/go/src/gopkg.in/redis.v3/commands.go:58 +0x34
gopkg.in/redis%2ev3.(*commandable).Auth(0xc8207eaa38, 0xc820280710, 0x10, 0xc8207eaa30)
	/home/sergey/go/src/gopkg.in/redis.v3/commands.go:65 +0x19d
gopkg.in/redis%2ev3.(*conn).init(0xc8209681e0, 0xc8201f4500, 0x0, 0x0)
	/home/sergey/go/src/gopkg.in/redis.v3/conn.go:52 +0x139
gopkg.in/redis%2ev3.newConnDialer.func1(0xc8201ead00, 0x0, 0x0)
	/home/sergey/go/src/gopkg.in/redis.v3/conn.go:36 +0x237
gopkg.in/redis%2ev3.(*connPool).new(0xc8202c8300, 0x25008471, 0x0, 0x0)
	/home/sergey/go/src/gopkg.in/redis.v3/pool.go:205 +0x164
gopkg.in/redis%2ev3.(*connPool).Remove(0xc8202c8300, 0xc8208a8f00, 0x0, 0x0)
	/home/sergey/go/src/gopkg.in/redis.v3/pool.go:259 +0x3d
gopkg.in/redis%2ev3.(*baseClient).putConn(0xc8201ead20, 0xc8208a8f00, 0x7f61d2cc9900, 0xc820ad43c0)
	/home/sergey/go/src/gopkg.in/redis.v3/redis.go:32 +0x252
gopkg.in/redis%2ev3.(*baseClient).process(0xc8201ead20, 0x7f61d0c29250, 0xc8206e2dc0)
	/home/sergey/go/src/gopkg.in/redis.v3/redis.go:73 +0x370
gopkg.in/redis%2ev3.(*baseClient).(gopkg.in/redis%2ev3.process)-fm(0x7f61d0c29250, 0xc8206e2dc0)
	/home/sergey/go/src/gopkg.in/redis.v3/redis.go:185 +0x34
gopkg.in/redis%2ev3.(*commandable).Process(0xc8201f38a8, 0x7f61d0c29250, 0xc8206e2dc0)
	/home/sergey/go/src/gopkg.in/redis.v3/commands.go:58 +0x34
gopkg.in/redis%2ev3.(*commandable).EvalSha(0xc8201f38a8, 0xc8201bdbc0, 0x28, 0xc820ab7da8, 0x1, 0x1, 0xc820ab7d98, 0x1, 0x1, 0xc81ffc90b9)
```

**Comment by vmihailenco** [comment]
Thanks for the report. This bug is specified to slow connections with `MaxRetries > 0`. Should be fixed with https://github.com/go-redis/redis/pull/173

**Comment by vmihailenco** [comment]
Fixed in v3.2.12. Again thanks for the report.
